File: platform_src/android/app/src/main/java/org/elastos/trinity/runtime/didsessions/SignInOptions.java
package org.elastos.trinity.runtime.didsessions;

import org.json.JSONException;
import org.json.JSONObject;

public class SignInOptions {
    public String sessionLanguage = null;

    public SignInOptions() {
    }

    public JSONObject asJsonObject() {
        try {
            JSONObject jsonObj = new JSONObject();
            jsonObj.put("sessionLanguage", sessionLanguage);
            return jsonObj;
        } catch (JSONException e) {
            e.printStackTrace();
            return null;
        }
    }

    public static SignInOptions fromJsonObject(JSONObject jsonObj) {
        try {
            SignInOptions options = new SignInOptions();

            if (jsonObj.has("sessionLanguage"))
                options.sessionLanguage = jsonObj.getString("sessionLanguage");

            return options;
        } catch (JSONException e) {
            e.printStackTrace();
            return null;
        }
    }
}

File: platform_src/electron/main/didsessions/SignInOptions.ts
export class SignInOptions {
    public sessionLanguage: string = null;

    constructor() {
    }

    public asJsonObject(): any {
        let jsonObj: any = {};
        jsonObj.sessionLanguage = this.sessionLanguage;
        return jsonObj;
    }

    public static fromJsonObject(jsonObj: any): SignInOptions {
        let options: SignInOptions = new SignInOptions();
        
        if (jsonObj.sessionLanguage != null)
            options.sessionLanguage = jsonObj.sessionLanguage as string;

        return options;
    }


}